pkg/models/canonical-form.go

```diff
@@ -0,0 +1,167 @@
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// Routes represents a slice of Routes
+type Routes []Route
+
+// Route captures the key elements that define a Route in a string that maps to a URL structure. These values
+// are captured as runtime routes, meaning that if the CF Application manifest is configured to disable all routes
+// with the `no-route` value, it will translate into an empty slice.
+// By default Cloud Foundry attempts to create a route for each application unless the `no-route` field is set to true.
+// For further details check: https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#no-route
+// and https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#random-route
+// Example
+// ---
+//
+//	...
+//	routes:
+//	- route: example.com
+//	  protocol: http2
+//	- route: www.example.com/foo
+//	- route: tcp-example.com:1234
+type Route struct {
+	// URL captures the FQDN, path and port of the route.
+	URL string `json:"url"`
+	// Protocol captures the protocol type: http, http2 or tcp. Note that the CF `protocol` field is only available
+	// for CF deployments that use HTTP/2 routing.
+	Protocol RouteProtocol `json:"protocol"`
+}
+
+type RouteProtocol string
+
+const (
+	HTTP  RouteProtocol = "http"
+	HTTPS RouteProtocol = "https"
+	TCP   RouteProtocol = "tcp"
+)
+
+// Services represents a slice of Service
+type Services []Service
+
+// Service contains the specification for an existing Cloud Foundry service required by the application.
+// Examples:
+// ---
+//
+//	...
+//	services:
+//	  - service-1
+//	  - name: service-2
+//	  - name: service-3
+//	    parameters:
+//	      key-1: value-1
+//	      key-2: [value-2, value-3]
+//	      key-3: ... any other kind of value ...
+//	  - name: service-4
+//	    binding_name: binding-1
+type Service struct {
+	// Name represents the name of the Cloud Foundry service required by the application. This field
+	// represents the runtime name of the service, captured from the 3 different cases where
+	// the service name can be listed.
+	// For more information check https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#services-block
+	Name string `json:"name"`
+	// Parameters contain the k/v relationship for the aplication to bind to the service
+	Parameters map[string]interface{} `json:"parameters,omitempty"`
+}
+
+// Processes represents a slice of Processes.
+type Processes []Process
+
+// Process represents the abstraction of the specification of a Cloud Foundry Process.
+// For more information check https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#processes
+type Process struct {
+	// Type captures the `type` field in the Process specification. Accepted values are `web` or `worker`
+	Type ProcessType `json:"type,omitempty"`
+	// Image represents the pull spec of the container image.
+	Image string `json:"image"`
```

Comment:
We how have a `Docker` in the application and we don't have Image at the Process level anymore. Just to align verbatim with the source rather than try to interpret it.